app/controllers/volunteers_controller.rb

```diff
@@ -0,0 +1,32 @@
+class VolunteersController < ApplicationController
+  before_action :set_volunteer, only: %i[destroy]
+  before_action :find_post, only: %i[create destroy]
+
+  def create
+    @volunteer = @post.volunteers.create(volunteer_params)
+    @volunteer.user = current_user
+
+    if @volunteer.save
+      flash[:notice] = 'You have successfully applied'
+      redirect_back(fallback_location: root_path)
+    else
+      flash.now[:alert] = 'Could not apply, try again'
```

Comment:
Show proper message, because of which user can apply.

Reply:
I can't think of any specific reason why this could fail. This why I have just written 'Try again'